ringpop.go

```diff
@@ -213,6 +233,20 @@ func (rp *Ringpop) RegisterListener(l EventListener) {
 	rp.listeners = append(rp.listeners, l)
 }
 
+// getState gets the state of the current Ringpop instance.
+func (rp *Ringpop) getState() state {
+	rp.stateMutex.Lock()
+	defer rp.stateMutex.Unlock()
```

Comment:
Here is a blog post with some benchmarking results: http://lk4d4.darth.io/posts/defer/